### PR TITLE
#7 add empty value from zod spec

### PR DESCRIPTION
**Pull request by varnigithub**
- implemented empty fn and tests for 

> - zodnumber,
> - zodstring, 
> - zodboolean, 
> - zodobject, 
> - zodarray

### CODE REVIEW

File: experimental/fake/src/empty.test.ts [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+import { empty } from './empty'
+
+describe('empty value', () => {
+  test('primitives', () => {
+    expect(empty(z.number())).toBe(0)
+    expect(empty(z.string())).toBe('')
+    expect(empty(z.boolean())).toBe(false)
+  })
+  test('array', () => {
+    expectTypeOf(empty(z.number().array())).toMatchTypeOf<number[]>()
```
Comment (pervezfunctor) [inline experimental/fake/src/empty.test.ts]: add value test too

File: experimental/fake/src/empty.test.ts [hunk 2]
```diff
@@ -0,0 +1,25 @@
+import { describe, test, expectTypeOf, expect } from 'vitest'
+import { z } from 'zod'
+import { empty } from './empty'
+
+describe('empty value', () => {
+  test('primitives', () => {
+    expect(empty(z.number())).toBe(0)
+    expect(empty(z.string())).toBe('')
```
Comment (pervezfunctor) [inline experimental/fake/src/empty.test.ts:8]: add tests for undefined, null, Date etc

File: experimental/fake/src/empty.test.ts [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+    expect(empty(z.boolean())).toBe(false)
+  })
+  test('array', () => {
+    expectTypeOf(empty(z.number().array())).toMatchTypeOf<number[]>()
+    expectTypeOf(empty(z.string().array())).toMatchTypeOf<string[]>()
+    expectTypeOf(empty(z.boolean().array())).toMatchTypeOf<boolean[]>()
+  })
+  test('object', () => {
+    const Foo = z.object({ x: z.number(), y: z.string(), z: z.boolean() })
```
Comment (pervezfunctor) [inline experimental/fake/src/empty.test.ts]: test for nested objects, nested objects and arrays, nested arrays etc

File: experimental/fake/src/empty.ts [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+  if (type === 'ZodArray') {
+    return pipe(
+      range(0, faker.number.int({ max: 20 })),
+      map(() => empty(spec._def.type)),
+      toArray,
+    )
+  }
+
+  return undefined
+}
```
Comment (pervezfunctor) [inline experimental/fake/src/empty.ts:139]: add support for all of the types in supportedTypes.
